arviz/rcparams.py

```diff
@@ -93,6 +93,18 @@ def _validate_boolean(value):
     return value is True or value == "true"
 
 
+def _validate_tuple(value):
+    """Validate value is a tuple of size 2."""
+    if value is None:
+        return value
+    elif value == "auto":
+        return value
+    elif isinstance(value, tuple) and len(value) == 2:
+        return value
+    else:
+        raise ValueError("Only tuple of size 2 is valid.")
```

Comment:
Hi, I understand we need to validate a tuple of floats of length 2 right? There are many extra things to take into account. 

To begin with, the validation function must work with python objects (used at initialization from `defaultParams` and while using `rc` argument in `rc_context`) but also with _strings_ (used at initializaiton from arvizrc files and while using the `filename` argument in `rc_context`). 

I also think that the validation functions should be modularized as much as possible.

My suggestion is to divide the function into several smaller pieces and to make it much more restrictive. I would create some function like [`_listify_validator`](https://github.com/matplotlib/matplotlib/blob/c58ca603189f45b7e927e72b1ca6bb640bb9d376/lib/matplotlib/rcsetup.py#L74) from matplotlib. This function could be combined with `validate_float` to validate tuples/lists of floats of any length, creating a `validate_float_iterable` function. Eventually, a `validate_float_pair` could be created, or maybe to be more general a `length` argument can be added to `validate_float_iterable`.

Reply:
Yes this will make it much more sustainable. I'll make the changes, thanks!

Reply:
> Hi, I understand we need to validate a tuple of floats of length 2 right? There are many extra things to take into account.
> 
> To begin with, the validation function must work with python objects (used at initialization from `defaultParams` and while using `rc` argument in `rc_context`) but also with _strings_ (used at initializaiton from arvizrc files and while using the `filename` argument in `rc_context`).

How should I deal with compatibility with objects?

Reply:
The current function only works with objects. One example of working with objects and strings is [here](https://github.com/arviz-devs/arviz/blob/eb45dfb54d03da5e64690ac2cf0cb63983030377/arviz/rcparams.py#L30), where both `None` and `"none"` are accepted. In general, this is achieved by converting to the desired `type` and then make any extra validation assuming/knowing the type of the value (a good example is [`validate_probability`](https://github.com/arviz-devs/arviz/blob/eb45dfb54d03da5e64690ac2cf0cb63983030377/arviz/rcparams.py#L81), first we make sure the value is compatible with being a float and make sure its type is float and then its range is checked. I hope this answers the question.

Reply:
Yeah, thanks for the help! Actually I thought that this function wasn't compatible with objects. Now the confusion is cleared.